clang_delta/LocalToGlobal.cpp

```diff
@@ -208,6 +208,12 @@ bool LToGASTVisitor::makeLocalAsGlobalVar(FunctionDecl *FD,
 
   GlobalVarStr += ";\n";
 
+  auto& TheRewriter = ConsumerInstance->TheRewriter;
+  for (DeclContext* DC = FD; DC; DC = DC->getParent()) {
+    if (DC->getParent() && DC->getParent()->isTranslationUnit()) {
+      return TheRewriter.InsertTextBefore(cast<Decl>(DC)->getLocStart(), GlobalVarStr);
```

Comment:
Clang's ```Rewriter::InsertTextBefore``` returns ```false``` on successfully rewriting the source :) So, we have to change this line to:

    TheRewriter.InsertTextBefore(cast<Decl>(DC)->getLocStart(), GlobalVarStr);
    return true;

Otherwise, ```RecursiveASTVisitor``` would terminate early without visiting other nodes.